.github/workflows/ci-home.yml

```diff
@@ -22,16 +22,31 @@ on:
   workflow_dispatch:
 
 jobs:
+  generate_matrix:
+    timeout-minutes: 5
+    runs-on: ubuntu-24.04
+    outputs:
+      matrix: ${{ steps.generator.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+          cache-dependency-path: 'go.sum'
+      - name: Install gh-action-escape
+        run: curl -fsSL https://raw.githubusercontent.com/kachick/gh-action-escape/main/scripts/install-in-github-action.sh | sh -s v0.2.0
+      - name: Generate Matrix
+        id: generator
+        run: |
+          go run ./cmd/gen_matrix -event_name '${{ github.event_name }}' | gh-action-escape -name=matrix | tee -a "$GITHUB_OUTPUT"
   home-manager:
+    needs: [generate_matrix]
```

Comment:
I can omit this annoy way if GitHub provides https://github.com/actions/runner/issues/662